main.rb

```diff
@@ -38,75 +42,90 @@ def my_select
     array
   end
 
-  def my_all?(_argv = 0)
+  def my_all?(argv = nil)
     i = 0
-    j = 0
     array = []
     while i < size
       if block_given?
         return false unless yield(self[i])
-
-        j += 1
-        array.push(self[i])
-        return true if j == size
       else
-        return false unless self[i]
-
-        j += 1
+        if argv.is_a? Class
+          return true if is_a? argv
+        elsif argv.is_a? Regexp
+          return false unless self[i].match(argv)
+        elsif argv.nil?
+          return false unless self[i]
+        end
         array.push(self[i])
-        return true if j == size
       end
       i += 1
     end
-    return true if array.empty?
+    return false unless array.empty?
+
+    true
   end
 
-  def my_any?(_argv = 0)
+  def my_any?(argv = nil)
     i = 0
     array = []
-    size.times do
+    while i < size
       if block_given?
-        return false unless yield(self[i])
-
-        array.push(self[i])
-        return true
+        return true if yield(self[i])
       else
-        return false unless self[i]
-
+        if argv.is_a? Class
+          return true if self[i].is_a? argv
+        elsif argv.is_a? Regexp
+          return false unless self[i].match(argv)
+        elsif self[i]
+          return true
+        end
         array.push(self[i])
-        return true
       end
+      i += 1
     end
-    return true if array.empty?
+    return false if array.empty?
+
+    false
   end
 
-  def my_none?(_argv = 0)
+  def my_none?(argv = nil)
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given
```ruby
true_array = [nil, false, true, []] 
false_array = [nil, false, nil, false]
false_array.my_none? == true #true
true_array.my_none? == false #false
```